### DISCUSSION TITLE
JF SIP

### DISCUSSION THREAD

**Discussion by itsyourap** [JFC-Group/JF-Customisation#15] [Q&A]
Please read the [SIP Breakdown of JF](https://github.com/JFC-Group/JF-Customisation/blob/master/Instructions/SIP.md) and tell me if you have any way to play AMR/AMR-WB audio realtime and realtime conversion of audio captured from microphone to AMR/AMR-WB audio and send it via RTP server in the JioFiber Router and proceed with communication via SIP.

**Comment by manojmth** [comment]
Wonderful, great work on this! Had a suggestion, how about using opensips server (which would sit between the juice server and the client) to modify the headers as needed (can be done with the regex module) and using portsip sip client on pc/mobile (this already has the amr and amr-wb codecs)?

I might also try with this when I get some time. One could also use asterisk with the evs patch https://github.com/traud/asterisk-evs but then one would also need to patch the source code to modify the contact header as needed.

**Comment by manojmth** [comment]
Thanks be to God! Was able to successfully use opensips with microsip ua to connect and talk (outgoing call from microsip to a mobile no), clear audio both ways, all sip signalling working.

Used talend api tester (chrome extension) for initial api requests (device add request and otp confirmation)
Put in random imei and and my reg jio mobile no in msisdn for this request
https://github.com/JFC-Group/JF-Customisation/blob/92f71fbdd8f388b57417a9c55063fc88851c1610/Research/SIP.md?plain=1#L47 

After getting the xml, setup opensips with uac_registrant and modified all needed headers through config script. Used topology hiding as well. As opensips works, kamailio should as well, as they are nearly the same.

Packet capture is not working on my router, don't know if it's because of updated firmware or diff model. Is something needed to be done before sending this?
https://github.com/JFC-Group/JF-Customisation/blob/92f71fbdd8f388b57417a9c55063fc88851c1610/Research/SIP.md?plain=1#L33
I get a bad request response, only request_account and request_mac are working for me. 

Haven't got incoming from the jio landline working, no incoming invite at the opensips server which is registered properly. Could you please use packet capture if it still works for you and check what happens during an incoming call when using jiojoin?

Now that jiofiber can be used without the gateway for internet connectivity https://broadbandforum.co/threads/successful-authentication-tplink-ont-without-original-jiofiber.221140/
would be good if the juice server application on the gateway can be decompiled and one can use an alt server like opensips to directly talk to the jio backend voip server, masquerading as the juice server.

**Reply by itsyourap** [reply]
Sorry for the late response!
Yes! Indeed, in the last firmware update, Jio has disabled the packet capture feature. We can still packet capture via the JioJoin app (if you modify some of the methods during runtime).

It is great to hear you have configured OpenSips server. I wish I were a pro in Linux but I am not, so I don't know much about that. Anyways, if there is a way to setup OpenSips in Windows, let me know. 

Until then, I wish you the best

**Reply by new-creation7** [reply]
Hi, that's interesting 

Can you please share how to 
> setup opensips with uac_registrant and modified all needed headers through config script. Used topology hiding as well.

**Reply by new-creation7** [reply]
> Sorry for the late response!
> Yes! Indeed, in the last firmware update, Jio has disabled the packet capture feature. We can still packet capture via the JioJoin app (if you modify some of the methods during runtime).
> 
> It is great to hear you have configured OpenSips server. I wish I were a pro in Linux but I am not, so I don't know much about that. Anyways, if there is a way to setup OpenSips in Windows, let me know. 
> 
> Until then, I wish you the best

https://ipv6.rs/tutorial/Windows_10/openSIPS/

Thank you

**Reply by itsyourap** [reply]
> > Sorry for the late response!
> > Yes! Indeed, in the last firmware update, Jio has disabled the packet capture feature. We can still packet capture via the JioJoin app (if you modify some of the methods during runtime).
> > It is great to hear you have configured OpenSips server. I wish I were a pro in Linux but I am not, so I don't know much about that. Anyways, if there is a way to setup OpenSips in Windows, let me know.
> > Until then, I wish you the best
> 
> [ipv6.rs/tutorial/Windows_10/openSIPS](https://ipv6.rs/tutorial/Windows_10/openSIPS/)
> 
> Thank you

The tutorial mention getting the "Windows Binary Packages" from the opensips website but it is nowhere.

**Reply by new-creation7** [reply]
Yes, there is no package for Windows, My apologies.

**Reply by itsyourap** [reply]
https://github.com/JFC-Group/JF-Customisation/discussions/70#discussioncomment-11953542

**Comment by pramodhrachuri** [comment]
Were you able to use apps like linphone?
I remember it working for my BSNL FTH connection back in the days. If we can figure that out, I think I will be done with my VPN setup.

**Reply by itsyourap** [reply]
You need to recompile linphone with some changes to make it work. You can read [SIP Breakdown](https://github.com/JFC-Group/JF-Customisation/blob/master/Instructions/SIP.md) to understand what you need to change.

**Reply by pramodhrachuri** [reply]
Thanks for the response. 

Unfortunately I don't live in the same place to get the password phrase. I've a wireguard VPN that allows me connect to home but JioJoin doesn't let me connect.

Any other way to get the passphrase?

**Reply by itsyourap** [reply]
https://github.com/JFC-Group/JF-Customisation/discussions/70#discussioncomment-11953542

**Comment by itsyourap** [comment]
Transferred to #70

**Comment by codenamekraken** [comment]
can any one will help how to use the jio sip service

**Reply by itsyourap** [reply]
https://github.com/JFC-Group/JF-Customisation/discussions/70#discussioncomment-11953542
